guides/plugins/plugins/storefront/add-listing-filters.md

```diff
@@ -0,0 +1,174 @@
+
+{% page-ref page="./../plugin-fundamentals/listening-to-events.md" %}
+
+As you're familiar with other customization, we'll start creating this new class in the same path as you're seeing in
+Shopware's core - `/src/Subscriber/ExampleListingSubscriber.php`.
+
+New listing filters, e.g. for your product listing, can be registered via the event `\Shopware\Core\Content\Product\Events\ProductListingCollectFilterEvent`
+This event was introduced to enable every developer to specify the metadata for a filter. The handling, meaning if and 
+how a filter is added, is done by Shopware's core:
+
+```php
+    public static function getSubscribedEvents()
+    {
+        return [
+            ProductListingCollectFilterEvent::class => 'addFilter'
+        ];
+    }
+```
+
+After that, you can start to actually add your custom filters. Arguably an important step is to define your filter.
+Therefore, you're able to use the `Filter` class, including the parameters below:
+
+| Parameter | Description |
+| --- | --- | 
+| `name` | Unique name of the filter |
+| `filtered` | Set this option to `true` if this filter is active |
+| `aggregations` | Defines aggregations behind a filter. Sometimes a filter contains multiple aggregations like properties |
+| `filter` | Sets the DAL filter which should be added to the criteria    |
+| `values` | Defines the values which will be added as `currentFilter` to the result |
+| `exclude` | Configure exclusions |
+
+As a result, an example filter could look like this:
+
+```php
+$filter = new Filter(
+    // name
+    'manufacturer',
+    
+    // filtered
+    !empty($ids),
+    
+    // aggregations
+    [new EntityAggregation('manufacturer', 'product.manufacturerId', 'product_manufacturer')],
+    
+    // filter
+    new EqualsAnyFilter('product.manufacturerId', $ids),
+    
+    // values
+    $ids
+);
+```
+
+So you get the existing filters by `ProductListingCollectFilterEvent`, define your new custom filter and add it to the
+existing ones. Here is a complete example implementation, implementing a filter on the product information `isCloseout`.
+please note the comments for explanation:
+
+{% code title="<plugin root>/src/Subscriber/ExampleListingSubscriber.php" %}
+```php
+class ExampleListingSubscriber implements EventSubscriberInterface
+{
+    // register event
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ProductListingCollectFilterEvent::class => 'addFilter'
+        ];
+    }
+
+    public function addFilter(ProductListingCollectFilterEvent $event)
+    {
+        // fetch existing filters
+        $filters = $event->getFilters();
+        $request = $event->getRequest();
+
+        $filtered = (bool) $request->get('isCloseout');
+
+        $filter = new Filter(
+            // unique name of the filter
+            'isCloseout',
+            
+            // defines if this filter is active
+            $filtered === true,
```

Comment:
This check is unnecessery :smile: 

You've just casted $filtered to be bool.

```suggestion
            $filtered,
```

Reply:
Yeah, possibly, I was referring to the core at this point 🙈